### ISSUE TITLE
training failed, when only using REA trick with probability 0.5

### ISSUE THREAD

**Issue by maliho0803** [michuanhaohao/reid-strong-baseline#54]
baseline-s training success.
baseline-s + warmup training success.
baseline-s + REA training failed, the loss will not decrease when reach about 6~7
Epoch[21] Iteration[20/183] Loss: 7.762, Acc: 0.000, Base Lr: 3.50e-04
2019-07-15 15:43:30,428 reid_baseline.train INFO: Epoch[21] Iteration[40/183] Loss: 7.493, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:43:34,726 reid_baseline.train INFO: Epoch[21] Iteration[60/183] Loss: 7.322, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:43:39,028 reid_baseline.train INFO: Epoch[21] Iteration[80/183] Loss: 7.168, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:43:43,340 reid_baseline.train INFO: Epoch[21] Iteration[100/183] Loss: 7.039, Acc: 0.003, Base Lr: 3.50e-04
2019-07-15 15:43:47,659 reid_baseline.train INFO: Epoch[21] Iteration[120/183] Loss: 6.964, Acc: 0.003, Base Lr: 3.50e-04
2019-07-15 15:43:51,976 reid_baseline.train INFO: Epoch[21] Iteration[140/183] Loss: 6.866, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:43:56,293 reid_baseline.train INFO: Epoch[21] Iteration[160/183] Loss: 6.767, Acc: 0.004, Base Lr: 3.50e-04
2019-07-15 15:44:00,601 reid_baseline.train INFO: Epoch[21] Iteration[180/183] Loss: 6.629, Acc: 0.005, Base Lr: 3.50e-04
2019-07-15 15:44:01,318 reid_baseline.train INFO: Epoch 21 done. Time per batch: 39.010[s] Speed: 300.2[samples/s]
2019-07-15 15:44:01,318 reid_baseline.train INFO: ----------
2019-07-15 15:44:06,187 reid_baseline.train INFO: Epoch[22] Iteration[20/183] Loss: 7.096, Acc: 0.000, Base Lr: 3.50e-04
2019-07-15 15:44:10,510 reid_baseline.train INFO: Epoch[22] Iteration[40/183] Loss: 7.081, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:44:14,827 reid_baseline.train INFO: Epoch[22] Iteration[60/183] Loss: 7.040, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:44:19,150 reid_baseline.train INFO: Epoch[22] Iteration[80/183] Loss: 6.981, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:44:23,461 reid_baseline.train INFO: Epoch[22] Iteration[100/183] Loss: 6.930, Acc: 0.003, Base Lr: 3.50e-04
2019-07-15 15:44:27,773 reid_baseline.train INFO: Epoch[22] Iteration[120/183] Loss: 6.866, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:44:32,091 reid_baseline.train INFO: Epoch[22] Iteration[140/183] Loss: 6.802, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:44:36,403 reid_baseline.train INFO: Epoch[22] Iteration[160/183] Loss: 6.718, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:44:40,717 reid_baseline.train INFO: Epoch[22] Iteration[180/183] Loss: 6.591, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:44:41,422 reid_baseline.train INFO: Epoch 22 done. Time per batch: 39.035[s] Speed: 300.0[samples/s]
2019-07-15 15:44:41,422 reid_baseline.train INFO: ----------
2019-07-15 15:44:46,276 reid_baseline.train INFO: Epoch[23] Iteration[20/182] Loss: 7.158, Acc: 0.000, Base Lr: 3.50e-04
2019-07-15 15:44:50,571 reid_baseline.train INFO: Epoch[23] Iteration[40/182] Loss: 7.092, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:44:54,876 reid_baseline.train INFO: Epoch[23] Iteration[60/182] Loss: 7.040, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:44:59,191 reid_baseline.train INFO: Epoch[23] Iteration[80/182] Loss: 6.985, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:45:03,499 reid_baseline.train INFO: Epoch[23] Iteration[100/182] Loss: 6.927, Acc: 0.003, Base Lr: 3.50e-04
2019-07-15 15:45:07,817 reid_baseline.train INFO: Epoch[23] Iteration[120/182] Loss: 6.867, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:45:12,124 reid_baseline.train INFO: Epoch[23] Iteration[140/182] Loss: 6.801, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:45:16,427 reid_baseline.train INFO: Epoch[23] Iteration[160/182] Loss: 6.728, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:45:20,734 reid_baseline.train INFO: Epoch[23] Iteration[180/182] Loss: 6.588, Acc: 0.003, Base Lr: 3.50e-04
2019-07-15 15:45:21,219 reid_baseline.train INFO: Epoch 23 done. Time per batch: 38.763[s] Speed: 300.5[samples/s]
2019-07-15 15:45:21,219 reid_baseline.train INFO: ----------
2019-07-15 15:45:26,122 reid_baseline.train INFO: Epoch[24] Iteration[20/183] Loss: 7.328, Acc: 0.000, Base Lr: 3.50e-04
2019-07-15 15:45:30,441 reid_baseline.train INFO: Epoch[24] Iteration[40/183] Loss: 7.210, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:45:34,763 reid_baseline.train INFO: Epoch[24] Iteration[60/183] Loss: 7.120, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:45:39,080 reid_baseline.train INFO: Epoch[24] Iteration[80/183] Loss: 7.024, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:45:43,397 reid_baseline.train INFO: Epoch[24] Iteration[100/183] Loss: 6.967, Acc: 0.001, Base Lr: 3.50e-04
2019-07-15 15:45:47,701 reid_baseline.train INFO: Epoch[24] Iteration[120/183] Loss: 6.891, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:45:52,020 reid_baseline.train INFO: Epoch[24] Iteration[140/183] Loss: 6.814, Acc: 0.002, Base Lr: 3.50e-04
2019-07-15 15:45:56,327 reid_baseline.train INFO: Epoch[24] Iteration[160/183] Loss: 6.723, Acc: 0.003, Base Lr: 3.50e-04
2019-07-15 15:46:00,632 reid_baseline.train INFO: Epoch[24] Iteration[180/183] Loss: 6.530, Acc: 0.005, Base Lr: 3.50e-0

baseline-s + warmup + REA trains success.

do you find such problem?

**Comment by michuanhaohao** [comment]
Maybe warmup is important to the convergence of REA? We never try it. But I think REA can converge with another learning rate.

**Comment by shaoniangu** [comment]
Yes. The model may have difficulties in convergence if the warm up strategy is removed even without REA when using base lr=0.00035, let alone the model with REA. This phenomenon proves the effectiveness of warm up strategy. The model may converge using a smaller base lr.
